torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -627,7 +640,7 @@ void ProcessGroupNCCL::workCleanupLoop() {
     for (auto it = workList_.begin(); it != workList_.end();
          /* no increment*/) {
       auto& work = *it;
-      if (work->isCompleted()) {
+      if (work->isCompletedAndThrowException()) {
```

Comment:
I don't think its safe to throw an exception from here since we'll kill the thread and crash the entire process.

Reply:
We revised the solution from guard function registered by cudaStreamAddCallback, to throw exception in this cleanup thread. I guess here is intended to crash the entire process to let elastic launcher catch and resume.